flask/boxwise_flask/graph_ql/types.gql

```diff
@@ -0,0 +1,154 @@
+type Box {
+  id: ID!
+  boxLabelIdentifier: String!
+  location: Location!
+  items: Int!
+  product: Product!
+  gender: ProductGender
+  # A size from a size range, consider making this enum
+  size: String!
+  state: BoxState!
+  qrCode: QrCode!
+  createdBy: User!
+  createdOn: Datetime!
+  # The user who last changed something about the box
+  lastModifiedBy: User!
+  lastModifiedOn: Datetime!
+  comment: String
+}
+
+type QrCode {
+  id: ID!
+  code: String!
+  box: Box
+  createdOn: Datetime
+  # The user who generated QR code
+  createdBy: User!
+}
+
+type Product {
+  id: ID!
+  name: String!
+  category: ProductCategory!
+  sizeRange: SizeRange!
+  sizes: [String!]!
+  base: Base!
+  price: Float
+  gender: ProductGender
+  createdBy: User!
+  createdOn: Datetime!
+  # The user who last changed something about the box
+  lastModifiedBy: User!
+  lastModifiedOn: Datetime!
+}
+
+type ProductCategory {
+  id: ID!
+  name: String!
+  products: [Product]
+  sizeRanges: [SizeRange]
+  # Non-clothing categories don't have a product gender
+  hasGender: Boolean!
+}
+
+enum ProductGender {
+  Men
+  Women
+  UnisexAdult
+  UnisexChild
+  UnisexBaby
+  TeenGirl
+  TeenBoy
+  Girl
+  Boy
+}
+
+enum BoxState {
+  InStock
+  Lost
+  Ordered
+  Picked
+  Donated
+  Scrap
+}
+
+type SizeRange {
+  id: ID!
+  label: String!
+  sizes: [String!]!
+  productCategory: [ProductCategory!]
+}
+
+type Location {
+  id: ID!
+  base: Base!
+  name: String
+  isShop: Boolean!
+  isDeleted: Boolean!
+  # List of all the boxes in the location
+  boxes: [Box!]
+  boxState: BoxState
+  createdOn: Datetime!
+  # The user who generated QR code
+  createdBy: User!
+  lastModifiedBy: User!
+  lastModifiedOn: Datetime!
+}
+
+type Base {
+  id: ID!
+  name: String!
+  organisation: Organisation!
+  locations: [Location!]
+  currencyName: String
+}
+
+type Organisation {
+  id: ID!
+  name: String!
+  bases: [Base!]
+}
+
+type User {
+  id: ID!
+  organisation: Organisation
+  name: String
+  email: String!
+  validFirstDay: Date
+  validLastDay: Date
+  bases: [Base]
+  lastLogin: Datetime
+  lastAction: Datetime
+}
+
+input CreateBoxInput {
```

Comment:
@aerinsol I need your input about what fields could possibly be included in a `UpdateBoxInput`. All of `CreateBoxInput` but qrCode and createdBy?

Reply:
Not Updated
- id
- boxLabelIdentifier
- qrcode
- createdBy

Always Updated
- lastModifiedBy
- lastModifiedOn

Not yet dealt with in this implementation
- BoxState

Reply:
@pylipp again, we may want to change some of the behavior down the line for the seamless transfer functionality, but that's out of scope of this PR :)